Add CERNBox BYOA proposal (#460)
HSF/hsf.github.io: _gsocproposals/2019/proposal_CERNBox_BYOA.md

--
title: CERNBox: Bring Your Own Application
layout: gsoc_proposal
project: CERNBox
year: 2019
organization:
 - CERN
 - ownCloud
 - AARNet
---

## Description

Description
CERNBox is a sync and share collaborative cloud storage solution at CERN used by more than 16K users
and storing close to 7PB of data. The service has evolved from being a platform providing sync and share 
services to a collaboration hub which empowers end-users to perform work collaboratively. 

Collaboration happens thanks to the integration of the open source cloud storage sync and share (CS3) platform ownCloud and with a variety of application providers like Office 365, OnlyOffice and SWAN. Unfortunately, current
integrations with different application providers are difficult to be re-used across platforms and require a huge development effort to achieve the desired integration.

> Since the move to cloud based services the user freedom was lost. The user does not have anymore the possibility to chosse the applications she wants to use to perfrom their job; is the site administratgor that decides for her. The goal of this project is to provide a mechanism using a well-defined communication protocol to integrate any collaborative service with the ownCloud platform. For such purpose, the student will leverage on the existing CS3 APIs created at CERN and will focus on the integration of the new ownCloud Web interface (codenamed Phoenix) with the CERN REVA platform using Google gRPC and Google Protocol buffers technologies.

Reviewer: ```suggestion
Since the move to cloud based services the user freedom was lost. The users do not have anymore the possibility to chose the applications they want to use to perform their job; is the site administrator that decides for her. The goal of this project is to provide a mechanism using a well-defined communication protocol to integrate any collaborative service with the ownCloud platform. For such purpose, the student will leverage on the existing CS3 APIs created at CERN and will focus on the integration of the new ownCloud Web interface (code-named Phoenix) with the CERN REVA platform using Google gRPC and Google Protocol buffers technologies.
```

> Since the move to cloud based services the user freedom was lost. The user does not have anymore the possibility to chosse the applications she wants to use to perfrom their job; is the site administratgor that decides for her. The goal of this project is to provide a mechanism using a well-defined communication protocol to integrate any collaborative service with the ownCloud platform. For such purpose, the student will leverage on the existing CS3 APIs created at CERN and will focus on the integration of the new ownCloud Web interface (codenamed Phoenix) with the CERN REVA platform using Google gRPC and Google Protocol buffers technologies.

Reviewer: Several typos, I propose the genderless:
 The users do not have anymore the possibility to chose the applications they want to use to perform their job; it is the site administrator who decides for them. The goal of this project is to provide a mechanism using a well-defined communication protocol to integrate any collaborative service with the ownCloud platform. For such purpose, the student will leverage on the existing CS3 APIs created at CERN and will focus on the integration of the new ownCloud Web interface (code-named Phoenix) with the CERN REVA platform using Google gRPC and Google Protocol buffers technologies.
